stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/composables/AudioWaveSeekbar.kt

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (c) 2014-2023 Stream.io Inc. All rights reserved.
+ *
+ * Licensed under the Stream License;
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://github.com/GetStream/stream-chat-android/blob/main/LICENSE
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.getstream.chat.android.compose.ui.composables
+
+import androidx.compose.foundation.Canvas
+import androidx.compose.foundation.gestures.detectHorizontalDragGestures
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.geometry.CornerRadius
+import androidx.compose.ui.geometry.Offset
+import androidx.compose.ui.geometry.Rect
+import androidx.compose.ui.geometry.Size
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.drawscope.DrawScope
+import androidx.compose.ui.input.pointer.pointerInput
+import androidx.compose.ui.unit.dp
+
+private const val DEFAULT_TRACKER_HEIGHT_DP = 60
+private const val DEFAULT_TRACKER_WIDTH = 4
+
+@Composable
+public fun AudioWaveVSeekbar(
+    waveBars: List<Float>,
+    modifier: Modifier = Modifier,
+    colorLeft: Color = Color.Blue,
+    colorRight: Color = Color.Gray,
+    trackerDraw: DrawScope.(Float, Size) -> Unit = { progressWidth, trackerSize ->
+        drawRoundRect(
+            color = Color.Red,
+            topLeft = Offset(
+                java.lang.Float.min(
+                    java.lang.Float.max(progressWidth - trackerSize.width / 2, 0F),
+                    size.width - trackerSize.width
+                ),
+                0F
+            ),
+            size = trackerSize,
+            cornerRadius = CornerRadius(10F, 10F)
+        )
+    },
+) {
+
+    var progress by remember { mutableStateOf(0F) }
+    var isDragging by remember { mutableStateOf(false) }
+
+    var seekWidth: Float? = null
+    val barSpacing = 0.4
+
+    Canvas(
+        modifier = modifier.pointerInput(Unit) {
+            detectHorizontalDragGestures(
+                onDragStart = {
+                    isDragging = true
+
+                    progress = it.x * 100F / (seekWidth ?: 10F)
```

Comment:
Would be good extracting "magic numbers" and adding javadocs to this class, but it could be covered on a different PR